File: class2relational/src/main/java/org/atlanmod/model/dynamic/relationalModel/TableToColumns.java
package org.atlanmod.model.dynamic.relationalModel;

import org.atlanmod.model.dynamic.DynamicElement;

import java.util.ArrayList;
import java.util.List;

public class TableToColumns extends RelationalLink {

    public TableToColumns(RelationalTable source, List<RelationalColumn> target){
        super(RelationalModel.TABLE_COLUMNS, source, null);
        this.setTarget(new ArrayList<>(target));
    }

    public String toString() {
        StringBuilder target_ids = new StringBuilder("[");
        for (int i=0; i<getTarget().size(); i++)
            target_ids.append(((RelationalElement) this.getTarget().get(i)).getId()).append(i == getTarget().size() - 1 ? "]" : ",");
        return "(" + ((RelationalElement) this.getSource()).getId() + ", " + super.getType() + ", " + target_ids + ")";
    }

}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/relationalmodel/TableToColumns.scala
package org.atlanmod.class2relational.model.relationalmodel

class TableToColumns (source: RelationalTable, target: List[RelationalColumn])
  extends RelationalLink(RelationalMetamodel.TABLE_COLUMNS, source, target){

    def this(source: RelationalTable, target: RelationalColumn) =
        this(source, List(target))

    override def getSource: RelationalTable = source
    override def getTarget: List[RelationalColumn] = target

    override def toString: String =
        "(" + source.getId + ", " + getType + ", " + target.map(a => a.getId).mkString("[", ", ", "]") + ")"

}
